include/pybind11/pytypes.h

```diff
@@ -890,6 +891,23 @@ class iterator : public object {
     object value = {};
 };
 
+class type : public object {
+public:
+    PYBIND11_OBJECT_DEFAULT(type, object, PyType_Check);
+
+    // Explicit omitted here since PyTypeObject required (rather than just PyObject)
+    type(PyTypeObject* type_ptr) : object((PyObject*) type_ptr, borrowed_t()) {}
```

Comment:
Do we want/need this? I'm a bit weary of having (too many) C API types in the pybind11 API.

But ofc, with the weird C-style inheritance, it might be useful, yes. Though forcing the user to just call `type((PyObject*) something)` might be just as good? It would push the C API ugliness to the user (who already chose to interact with the C API without pybind11?).